vignettes/profiling.Rmd

```diff
@@ -0,0 +1,68 @@
+---
+title: "Profiling Performance"
+author: "Thomas Lin Pedersen"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Profiling Performance}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+The performance of `gtable` is critical as it has the potential to slow down
+plot creation a lot. In order to continuously monitor the performance of gtable
+the following piece of code is used to generate a profiling and inspect it:
+
+```{r}
+library(ggplot2)
+library(profvis)
+
+p <- ggplot() + 
+  scale_x_continuous(limits = c(0, 1)) + 
+  scale_y_continuous(limits = c(0, 1))
+
+p_build <- ggplot_build(p)
+
+profile <- profvis(for (i in seq_len(100)) ggplot_gtable(p_build))
+
+profile
+```
+
+```{r, eval=FALSE, include=FALSE}
+saveRDS(profile, file.path('profiling', paste0(packageVersion('gtable'), '.rds')))
+```
+
+The use of an empty `ggplot2` ensures that the profiling is based on real-life
+use and includes complex gtable assembly. Profilings for old version are kept 
+for reference and can be accessed at the 
+[github repository](https://github.com/r-lib/gtable/tree/master/vignettes/profiling).
+Care should be taken in not comparing profiling results across versions, as 
+changes to code outside of `gtable` can have profound effect on the results.
+Thus, the intend of profiling is to identify bottlenecks in the implementation
+that are ripe for improvement, more then to quantify improvements to performance
+over time.
+
+## Performance focused changes across versions
+To keep track of changes focused on improving the performance of `gtable` they
+are summarised below:
+
+### v`r packageVersion('gtable')`
```

Comment:
I think should literally inline the version here, and mention roughly the performance yield.

Reply:
like, spell it out? or what do you mean with "literally inline version"?

Reply:
Like write `1.2.10` so it's fixed for all time, rather than changing when the gtable version changes.

Reply:
My idea was to have it fixed for older versions and computed for the latest so we don’t end up with mismatches because we forget to update the version heading when we send it to CRAN

Reply:
Ah ok, that makes sense.